bou.ke/monkey does not support Apple Silicon

The tests which uses `bou.ke/monkey` does not work on AMD64 environments:

- `pkg/filekit/filepath_test.go`
- `pkg/typgo/gotest_test.go`

```
$ ./typicalw t
test> go test -cover -timeout=30s ./internal/app ./pkg/envkit ./pkg/errkit ./pkg/filekit ./pkg/oskit ./pkg/tmplkit ./pkg/typapp ./pkg/typgen ./pkg/typgo ./pkg/typmock ./pkg/typrls
# bou.ke/monkey
../../go/pkg/mod/bou.ke/monkey@v1.0.2/replace.go:24:14: undefined: jmpToFunctionValue
ok  	github.com/typical-go/typical-go/internal/app	(cached)	coverage: 93.9% of statements
ok  	github.com/typical-go/typical-go/pkg/envkit	(cached)	coverage: 91.4% of statements
ok  	github.com/typical-go/typical-go/pkg/errkit	(cached)	coverage: 100.0% of statements
FAIL	github.com/typical-go/typical-go/pkg/filekit [build failed]
ok  	github.com/typical-go/typical-go/pkg/oskit	(cached)	coverage: 100.0% of statements
ok  	github.com/typical-go/typical-go/pkg/tmplkit	(cached)	coverage: 90.0% of statements
ok  	github.com/typical-go/typical-go/pkg/typapp	(cached)	coverage: 59.6% of statements
ok  	github.com/typical-go/typical-go/pkg/typgen	(cached)	coverage: 59.0% of statements
FAIL	github.com/typical-go/typical-go/pkg/typgo [build failed]
ok  	github.com/typical-go/typical-go/pkg/typmock	(cached)	coverage: 65.5% of statements
ok  	github.com/typical-go/typical-go/pkg/typrls	(cached)	coverage: 80.8% of statements
FAIL
exit status 2
exit status 2
```

Note: 
```
../../go/pkg/mod/bou.ke/monkey@v1.0.2/replace.go:24:14: undefined: jmpToFunctionValue
```

`jmpToFunctionValue` is implemented only in [monkey_amd64.go](https://github.com/bouk/monkey/blob/master/monkey_amd64.go) and [monkey_386.go](https://github.com/bouk/monkey/blob/master/monkey_386.go)

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,6 @@ module github.com/typical-go/typical-go
 go 1.17
 
 require (
-	bou.ke/monkey v1.0.2
 	github.com/bmatcuk/doublestar/v2 v2.0.4
 	github.com/fatih/color v1.9.0
 	github.com/golang/mock v1.4.4

--- a/pkg/filekit/filepath.go
+++ b/pkg/filekit/filepath.go
@@ -19,7 +19,7 @@ func MatchMulti(patterns []string, name string) bool {
 
 // FindDir find directory
 func FindDir(includes, excludes []string) (packages []string, err error) {
-	err = filepath.Walk(".", func(path string, info os.FileInfo, err error) error {
+	err = filepath.WalkDir(".", func(path string, info os.DirEntry, err error) error {
 		if MatchMulti(includes, path) && !MatchMulti(excludes, path) && info.IsDir() {
 			packages = append(packages, "./"+path)
 		}

--- a/pkg/filekit/filepath_test.go
+++ b/pkg/filekit/filepath_test.go
@@ -1,10 +1,9 @@
 package filekit_test
 
 import (
-	"path/filepath"
+	"os"
 	"testing"
 
-	"bou.ke/monkey"
 	"github.com/stretchr/testify/require"
 	"github.com/typical-go/typical-go/pkg/filekit"
 )
@@ -39,47 +38,52 @@ func TestMatchMulti(t *testing.T) {
 	}
 }
 
+func setupFindDirTest(t *testing.T) func(t *testing.T) {
+	dirs := []string{
+		"tmp/internal/app/infra",
+		"tmp/internal/app/domain/mylibrary/controller",
+		"tmp/internal/app/domain/mylibrary/repo",
+		"tmp/internal/app/domain/mylibrary/service",
+		"tmp/internal/app/domain/mybook/controller",
+		"tmp/internal/app/domain/mybook/repo",
+		"tmp/internal/app/domain/mybook/service",
+		"tmp/internal/app/generated/constructor",
+		"tmp/internal/app/generated/config",
+	}
+
+	for _, dir := range dirs {
+		err := os.MkdirAll(dir, os.ModePerm)
+		if err != nil && !os.IsExist(err) {
+			t.Skip(err.Error())
+		}
+	}
+
+	return func(t *testing.T) {
+		os.RemoveAll("tmp")
+	}
+}
+
 func TestFindDir(t *testing.T) {
-	defer monkey.Patch(filepath.Walk,
-		func(root string, walkFn filepath.WalkFunc) error {
-			dirs := []string{
-				"internal/app/infra",
-				"internal/app/domain",
-				"internal/app/domain/mylibrary",
-				"internal/app/domain/mylibrary/controller",
-				"internal/app/domain/mylibrary/service",
-				"internal/app/domain/mylibrary/repo",
-				"internal/app/domain/mybook/controller",
-				"internal/app/domain/mybook/service",
-				"internal/app/domain/mybook/repo",
-				"internal/app/generated",
-				"internal/app/generated/constructor",
-				"internal/app/generated/config",
-			}
-
-			for _, dir := range dirs {
-				walkFn(dir, &filekit.FileInfo{IsDirField: true}, nil)
-			}
-
-			return nil
-		},
-	).Unpatch()
+	teardownTest := setupFindDirTest(t)
+	defer teardownTest(t)
 
 	dirs, err := filekit.FindDir(
-		[]string{"internal/**/*"},
-		[]string{"internal/**/generated/**"},
+		[]string{"tmp/internal/**/*"},
+		[]string{"tmp/internal/**/generated/**"},
 	)
 	require.NoError(t, err)
 	require.Equal(t, []string{
-		"./internal/app/infra",
-		"./internal/app/domain",
-		"./internal/app/domain/mylibrary",
-		"./internal/app/domain/mylibrary/controller",
-		"./internal/app/domain/mylibrary/service",
-		"./internal/app/domain/mylibrary/repo",
-		"./internal/app/domain/mybook/controller",
-		"./internal/app/domain/mybook/service",
-		"./internal/app/domain/mybook/repo",
-		"./internal/app/generated",
+		"./tmp/internal/app",
+		"./tmp/internal/app/domain",
+		"./tmp/internal/app/domain/mybook",
+		"./tmp/internal/app/domain/mybook/controller",
+		"./tmp/internal/app/domain/mybook/repo",
+		"./tmp/internal/app/domain/mybook/service",
+		"./tmp/internal/app/domain/mylibrary",
+		"./tmp/internal/app/domain/mylibrary/controller",
+		"./tmp/internal/app/domain/mylibrary/repo",
+		"./tmp/internal/app/domain/mylibrary/service",
+		"./tmp/internal/app/generated",
+		"./tmp/internal/app/infra",
 	}, dirs)
 }

--- a/pkg/typgo/gotest_test.go
+++ b/pkg/typgo/gotest_test.go
@@ -36,7 +36,7 @@ func setupGoTest(t *testing.T) func(t *testing.T) {
 	return func(t *testing.T) {
 		os.RemoveAll("pkg1")
 		os.RemoveAll("pkg2")
-		os.RemoveAll("pkg")
+		os.RemoveAll("pkg_mock")
 	}
 }
 

--- a/pkg/typgo/gotest_test.go
+++ b/pkg/typgo/gotest_test.go
@@ -2,13 +2,10 @@ package typgo_test
 
 import (
 	"flag"
-	"path/filepath"
+	"os"
 	"testing"
 	"time"
 
-	"github.com/typical-go/typical-go/pkg/filekit"
-
-	"bou.ke/monkey"
 	"github.com/stretchr/testify/require"
 	"github.com/typical-go/typical-go/pkg/typgo"
 	"github.com/urfave/cli/v2"
@@ -22,15 +19,30 @@ func TestGoTest_NoPackages(t *testing.T) {
 	require.NoError(t, testProj.Execute(c))
 }
 
+func setupGoTest(t *testing.T) func(t *testing.T) {
+	dirs := []string{
+		"pkg1",
+		"pkg2",
+		"pkg_mock",
+	}
+
+	for _, dir := range dirs {
+		err := os.MkdirAll(dir, os.ModePerm)
+		if err != nil && !os.IsExist(err) {
+			t.Skip(err.Error())
+		}
+	}
+
+	return func(t *testing.T) {
+		os.RemoveAll("pkg1")
+		os.RemoveAll("pkg2")
+		os.RemoveAll("pkg")
+	}
+}
+
 func TestGoTest(t *testing.T) {
-	defer monkey.Patch(filepath.Walk,
-		func(root string, walkFn filepath.WalkFunc) error {
-			walkFn("pkg1", &filekit.FileInfo{IsDirField: true}, nil)
-			walkFn("pkg2", &filekit.FileInfo{IsDirField: true}, nil)
-			walkFn("pkg/service_mock", &filekit.FileInfo{IsDirField: true}, nil)
-			return nil
-		},
-	).Unpatch()
+	teardownTest := setupGoTest(t)
+	defer teardownTest(t)
 
 	c := &typgo.Context{
 		Context: cli.NewContext(nil, &flag.FlagSet{}, nil),

